modules/analysis/src/model.rs

```diff
@@ -0,0 +1,193 @@
+use parking_lot::RwLock;
+use petgraph::Graph;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::fmt;
+use std::sync::{Arc, Once};
+use trustify_entity::relationship::Relationship;
+use utoipa::ToSchema;
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct AnalysisStatus {
+    pub sbom_count: i32,
+    pub graph_count: i32,
+}
+impl fmt::Display for crate::model::AnalysisStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "graph_count {}", self.graph_count)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct DepNode {
+    pub purl: String,
+    pub name: String,
+    pub deps: Vec<DepNode>,
+}
+impl fmt::Display for crate::model::DepNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.name)
+    }
+}
+impl DepNode {
+    pub fn new(purl: String, name: String, contains: Vec<DepNode>) -> DepNode {
+        DepNode {
+            purl,
+            name,
+            deps: contains,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct PackageNode {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+}
+impl fmt::Display for PackageNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.name)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct AdvisoryGraphSummary {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+
+    pub ancestors: Vec<PackageNode>,
+}
+impl AdvisoryGraphSummary {
+    pub fn new(
+        purl: String,
+        name: String,
+        published: String,
+        document_id: String,
+        product_name: String,
+        product_version: String,
+        ancestors: Vec<PackageNode>,
+    ) -> AdvisoryGraphSummary {
+        AdvisoryGraphSummary {
+            purl,
+            name,
+            published,
+            document_id,
+            product_name,
+            product_version,
+            ancestors,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct AdvisoryDepSummary {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+    pub deps: Option<Vec<DepNode>>,
+}
+impl crate::model::AdvisoryDepSummary {
+    pub fn new(
```

Comment:
When a ctor has the same parameters as the struct fields, which are `pub`, I think there's more clarity in just constructing the struct directly, instead of via a function. Without inlay hints, it's hard to know which field is which position, using a positional arg fn.